### PR TITLE
Onboarding disclaimer

### LINKED ISSUE #4181: Onboarding v2 - disclaimer
As a map owner,
I want to see a new-design disclaimer every time my map is accessed,
so I can prominently display T&Cs

**Design file**: `terria-onboarding.sketch` in https://drive.google.com/open?id=1ihNj5hr9pXnJxhsP9S9sRqvxntd4RVbz

**Invision**: https://projects.invisionapp.com/share/VQWC0P9FC3D#/screens/408368056

Currently we use errors to show the notification message - will leave to developer discretion on how to best implement.

### Current
image

### New
image

### DoD:
- [ ] Unit test written (if applicable)
- [x] Works on mobile (or deliberately disabled for mobile)
- [x] Code/peer reviewed (if didn't pair)
- [ ] Relevant docs are updated
- [x] Previewable/deployed to dev (see separation of deployments)
- [x] Previewable/deployed to test (see separation of deployments)
- [x] (If a map-specific ticket) Issue linked to a release ticket

### Acceptance Criteria:
- [x] Disclaimer contents still loaded in from a TerriaMap
- [x] Disclaimer looks like new design
- [ ] ...(more, write if you run into need for clarification/checks)

### PR DESCRIPTION
**Pull request by reginapramesti**
### What this PR does

Fixes #4181 

Implemented a disclaimer component based on designs

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Comment by soyarsauce** [comment]
Linked to https://github.com/TerriaJS/terriajs/issues/4181

**Review by soyarsauce — CHANGES_REQUESTED** [review]
Hey @reginapramesti - couldn't get this to work locally without docs & trying to infer all the settings needed for disclaimerSettings and whatever setting mechanism you would have used in a TerriaMap, could you share your config optoins/params+how to set up a terriamap?

**Comment by reginapramesti** [comment]
Oh my gosh I just realised this PR was poorly made - initially I accidentally created a PR against master branch (which had all of the details and link to issue), closed that and opened a new one with no description 🤦 sorry about that

**Comment by soyarsauce** [comment]
Aw no! I figured it was strange there was no link to the issue which is why I just copied it in. Does the description now have everything you had written?

**Comment by reginapramesti** [comment]
Yup, it's all there now 👍

**Comment by reginapramesti** [comment]
Ok now it should in theory work when built in conjunction with https://github.com/TerriaJS/TerriaMap/pull/450

File: lib/Models/Terria.ts [hunk 1]
```diff
@@ -195,6 +196,7 @@ export default class Terria {
     locationSearchBoundingBox: undefined,
     googleAnalyticsKey: undefined,
     rollbarAccessToken: undefined,
+    globalDisclaimer: undefined,
```
[the remark this answers is not in the document]
Comment (reginapramesti) [reply lib/Models/Terria.ts:202]: This is used in the `index.js` file of the map 😬 did it this way cause that's how disclaimers were done in master (from what I observe)

File: lib/ReactViewModels/ViewState.ts [hunk 2]
```diff
@@ -64,6 +64,8 @@ export default class ViewState {
   @observable showWelcomeMessage: boolean = false;
   @observable selectedHelpMenuItem: string = "";
   @observable helpPanelExpanded: boolean = false;
+  @observable disclaimerSettings: any | undefined = undefined;
+  @observable disclaimerVisible: boolean = false;
```
[the remark this answers is not in the document]
Comment (reginapramesti) [reply lib/ReactViewModels/ViewState.ts:77]: The action is in `index.js` (like the above). `disclaimerVisible` is modified in `hideDisclaimer` action in `ViewState` which is called in `confirm` function in Disclaimer 😄 I've been trying to keep most of the way things work fairly similar to master (i.e. button presses, how disclaimer gets loaded in, etc). Not sure if it's ideal? 😬

File: lib/Models/Terria.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (soyarsauce) [reply lib/Models/Terria.ts:202]: yeah i see it now with the config + index.js usage! 👍

File: lib/ReactViewModels/ViewState.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (soyarsauce) [reply lib/ReactViewModels/ViewState.ts:77]: 👍 yeah all good - was missing the index.js context

**Pull request merged**